glide-core/src/client/value_conversion.rs

```diff
@@ -287,6 +288,46 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // Used by ZRANDMEMBER when the WITHSCORES arg is passed.
+        // The server response can be a nil value, an empty array, a flat array of member-score pairs, or a two-dimensional array of member-score pairs.
+        // The server response scores can be strings or doubles. The conversions we do here are as follows:
+        //
+        // - if the server returned nil, return nil
+        // - if the server returned an empty array, return an empty array
```

Comment:
does this ever happen? I think the commands will return nil, and we won't ever get an empty array

Reply:
From the redis docs: "Array reply: when the additional count argument is passed, the command returns an array of members, or an empty array when key doesn't exist."